### DISCUSSION TITLE
Is there plan to use fleet.rs as rust compiler?

### DISCUSSION THREAD

**Discussion by seonglae** [napi-rs/napi-rs#1246] [Ideas]
**Fleet.rs**  is `cargo build` and `cargo run` and `cargo build`

I failed to compile my `napi-rs` module by [`fleet.rs`](fleet.rs)([repo](https://github.com/dimensionhq/fleet)) instead of cargo. It is still impressively fast and interesting. I'm not sure fleet is stable enough to use as builder. However, I found there is no discussions about `fleet.rs`.

**Comment by messense** [comment]
> I failed to compile my napi-rs module by [fleet.rs](https://github.com/napi-rs/napi-rs/discussions/fleet.rs)([repo](https://github.com/dimensionhq/fleet)) instead of cargo

Can you share more information on this? How did you compile it? What error message did it give?

**Reply by seonglae** [reply]
@messense 
When I run `fleet build`

```log
❯ fleet build

📝 Generated Fleet Config
🚀 Fleet is ready!
   Compiling proc-macro2 v1.0.40
   Compiling quote v1.0.20
   Compiling memchr v2.5.0
   Compiling syn v1.0.98
   Compiling regex-syntax v0.6.27
   Compiling napi-sys v2.2.2
   Compiling my_module v0.0.0 
   Compiling thread_local v1.1.4
   Compiling aho-corasick v0.7.18
   Compiling regex v1.6.0
   Compiling napi-derive-backend v1.0.34
   Compiling ctor v0.1.22
   Compiling napi-derive v2.6.0
   Compiling napi v2.6.3
    Finished dev [unoptimized + debuginfo] target(s) in 2.96s
```

After that I run command `napi build --platform --release`
```log
❯ pnpm build
   Compiling proc-macro2 v1.0.40
   Compiling quote v1.0.20
   Compiling unicode-ident v1.0.2
   Compiling syn v1.0.98
   Compiling memchr v2.5.0
   Compiling regex-syntax v0.6.27
   Compiling once_cell v1.13.0
   Compiling napi-build v2.0.1
   Compiling convert_case v0.5.0
error: incorrect value `...` for codegen option `split-debuginfo` - one of supported split-debuginfo modes (`off`, `packed`, or `unpacked`) was expected

error: could not compile `quote` due to previous error
warning: build failed, waiting for other jobs to finish...
error: build failed
Internal Error: Command failed: cargo build --release
    at checkExecSyncError (node:child_process:826:11)
    at Object.execSync (node:child_process:900:15)
    at BuildCommand.<anonymous> (/Users/alan.jo/Downloads/hdgen/node_modules/.pnpm/@napi-rs+cli@2.10.0/node_modules/@napi-rs/cli/scripts/index.js:31968:26)
    at Generator.next (<anonymous>)
    at fulfilled (/Users/alan.jo/Downloads/hdgen/node_modules/.pnpm/@napi-rs+cli@2.10.0/node_modules/@napi-rs/cli/scripts/index.js:3490:58)
 ELIFECYCLE  Command failed with exit code 1.
```
Above error occurs. I'm not a Rust build expert, so I don't  if this order is the way to go.

**Reply by messense** [reply]
Can you try `CARGO=fleet pnpm build`?

**Reply by seonglae** [reply]
```log
❯ CARGO=fleet pnpm build

> napi build --platform --release

📝 Generated Fleet Config
🚀 Fleet is ready!
   Compiling proc-macro2 v1.0.40
   Compiling quote v1.0.20
   Compiling unicode-ident v1.0.2
   Compiling syn v1.0.98
   Compiling memchr v2.5.0
   Compiling regex-syntax v0.6.27
   Compiling once_cell v1.13.0
   Compiling convert_case v0.5.0
   Compiling napi-build v2.0.1
error: incorrect value `...` for codegen option `split-debuginfo` - one of supported split-debuginfo modes (`off`, `packed`, or `unpacked`) was expected

error: could not compile `convert_case` due to previous error
warning: build failed, waiting for other jobs to finish...
error: build failed
=> You are not using a `nightly` compiler. Run `rustup default nightly`.
Internal Error: Command failed: fleet build --release
    at checkExecSyncError (node:child_process:826:11)
    at Object.execSync (node:child_process:900:15)
    at BuildCommand.<anonymous> (/Users/alan.jo/Downloads/hdgen/node_modules/.pnpm/@napi-rs+cli@2.10.0/node_modules/@napi-rs/cli/scripts/index.js:31968:26)
    at Generator.next (<anonymous>)
    at fulfilled (/Users/alan.jo/Downloads/hdgen/node_modules/.pnpm/@napi-rs+cli@2.10.0/node_modules/@napi-rs/cli/scripts/index.js:3490:58)
```

This error occurs even after I changed to rustup nightly

**Reply by messense** [reply]
That issue could be a fleet issue, other users also encountered it: https://github.com/dimensionhq/fleet/issues/2#issuecomment-1108034647.

They may misunderstand the documentation that `split-debuginfo = '...'  # Platform-specific.` is a valid config. To me it looks like the documentation don't want to write out platform specific value for `split-debuginfo` not that `...` is a valid option.

**Reply by messense** [reply]
`...` is definitely not a valid option for `split-debuginfo`, see https://doc.rust-lang.org/rustc/codegen-options/index.html#split-debuginfo.

I suggest you open a issue in fleet's issue tracker. Meanwhile you can try build without `--release` to see if debug build works.

**Reply by seonglae** [reply]
`CARGO=fleet pnpm napi build --platform`  without release option build module successful. It runs fine either
But It's weird `...` valued is passing to option when `--release` option added to `napi build` command

**Reply by messense** [reply]
It’s fleet’s fault: https://github.com/dimensionhq/fleet/blob/9855e5fb9ce8d369e978386b69cefa7f0b13e103/src/core/config/cargo.rs#L113
